src/uriHandler.ts

```diff
@@ -25,17 +26,20 @@ export class UriEventHandler extends vscode.EventEmitter<vscode.Uri> implements
     return UriEventHandler.instance;
   }
 
-  public handleUri(uri: vscode.Uri) {
-    switch (uri.path) {
+  public async handleUri(uri: vscode.Uri) {
+    const { path } = uri;
+    switch (path) {
       case "/authCallback":
         logger.debug("Got authCallback URI, firing as Event", uri);
         this.fire(uri);
         break;
       case "/consume":
         vscode.commands.executeCommand("confluent.topic.consume.fromUri", uri);
         break;
-      default:
-        logger.warn("Got unexpected URI, ignoring", uri);
+      case "/projectScaffold":
+        projectScaffoldUri.fire(uri);
+        vscode.window.showErrorMessage("Unsupported URI path.");
```

Comment:
Let's also remove this error notification unless it's part of the `default` case